### ISSUE TITLE
Delete Deck doesn't update side bar

### ISSUE THREAD

**Issue by wrudnicki16** [wrudnicki16/brainhacker#13]
Not sure if I need to be connecting the index sidebar component and the show page component to some main page component so the decks will be in sync when I delete a deck.

**Comment by wrudnicki16** [comment]
I'm also rendering two index items of the same id for some reason, and everything going to and from the index/show page is all sorts of out of sync. I've seen the state at a point after rendering the show page (which takes in all decks as props) where it has one deck for the deck after it has been called and reduced to an "id: deck" format, and the other where it is still in "deck: deck" format. So thats interesting and maybe should be checked out more
<img width="1440" alt="screen shot 2018-02-04 at 10 36 50 pm" src="https://user-images.githubusercontent.com/6581671/35791315-1938b8d8-09fc-11e8-946b-4ada3e9ab801.png">

**Comment by wrudnicki16** [comment]
Not sure if componentWillReceiveProps would work but I haven't been able to get it to work.

**Comment by wrudnicki16** [comment]
Because in your show page your state shape was wrong so when you deleted the item with a certain id it would find that id in the database, but in your frontend action-ajax-reducer cycle it wasn't actually deleting the thing you wanted it to. This was solved partially by getting the state shape correct - changing the reducer's receive Deck response to { deckId: deck }. This dealt with the double show problem. Then the delete issue was dealt with by creating a removeDeck action thunk - which we didn't have before, so therefore there was no frontend deletion process happening before. With these two changes, everything was updating correctly on both the index component and show component. Then all we did was push to history using a .then() after createDeck and removeDeck.
